### DISCUSSION TITLE
Hyperparameter evolution - Ultralytics YOLOv8 Docs

### DISCUSSION THREAD

**Discussion by giscus** [ultralytics/ultralytics#2455] [Docs]
# Hyperparameter evolution - Ultralytics YOLOv8 Docs

https://docs.ultralytics.com/yolov5/tutorials/hyperparameter_evolution/

**Comment by Jixuan-Fan** [comment]
Is there a plan to support hyperparameter evoluation in YOLOv8?

**Reply by glenn-jocher** [reply]
@include5636 yes! We have a Ray Tune integration for hyperparameter evolution in YOLOv8. See https://docs.ultralytics.com/usage/hyperparameter_tuning/

**Reply by PraveenMNaik** [reply]
Does it support for YOLOv5 or to use Ray tune we need to consider YOLOv8

**Reply by PraveenMNaik** [reply]
@glenn-jocher.... can you pls let me know the procedure of plotting above mentioned visualize image.... i have evolved my model for several generations and have evolve.csv file...

**Reply by glenn-jocher** [reply]
@PraveenMNaik the hyperparameter evolution feature with Ray Tune is supported in YOLOv8. For YOLOv5, you can follow the hyperparameter evolution guide in the YOLOv5 documentation.

To visualize your hyperparameter evolution results from the `evolve.csv` file, you can use the provided plotting tools in the repository. Load your `evolve.csv` into a data analysis tool like Pandas in Python, and then you can create plots using libraries such as Matplotlib or Seaborn. The `evolve.csv` file contains the metrics and hyperparameters for each generation, which you can plot to analyze the performance trends over generations.

For detailed steps on how to plot these results, please refer to the visualization section in our documentation. If you need further assistance, feel free to open an issue in the YOLOv8 repo, and we'll be happy to help you out! 😊

**Comment by meeselizabeth** [comment]
I am doing the --yolov5m-seg.pt --epochs 10 --evolve 300 and it has been running for over 24 hours now. How long will this take?

**Reply by glenn-jocher** [reply]
@meeselizabeth the duration of hyperparameter evolution with YOLOv8 can vary widely depending on several factors such as the complexity of the model, the size of the dataset, the number of epochs, the number of generations in the evolution process, and the hardware you're using. Running evolution for 300 generations, as you've specified, can indeed be quite time-consuming, especially if you're using a model with segmentation capabilities like `--yolov5m-seg.pt` (which is a YOLOv5 model reference, but I understand you're referring to a similar YOLOv8 segmentation model).

For a process that involves 300 generations, it's not unusual for it to take several days, especially if you're not using high-end GPUs. Each generation involves training the model for the specified number of epochs (10 in your case), which can take a significant amount of time.

To speed up the process, consider using more powerful hardware, reducing the number of generations, or decreasing the number of epochs per generation. However, keep in mind that reducing the evolutionary search space may impact the quality of the optimized hyperparameters.

If you need further guidance on hyperparameter evolution with YOLOv8, please refer to the Ultralytics Docs for more detailed information.
